src/main/java/Duke.java

```diff
@@ -1,10 +1,155 @@
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Scanner;
+
 public class Duke {
+    private static Task[] records = new Task[100];
+    private static int recordsNum = 0;
+
+    public static String generateResponse(String input){
+
+        String trimmedInput = input.trim();
+        String endMessage = "Bye. Hope to see you again soon!";
+
+        if(trimmedInput.equals("bye")){
+            return endMessage;
+        }
+
+        if(trimmedInput.equals("list")){
+            printAllTasks();
+            return "";
+        }
+
+        int spaceIdx = trimmedInput.indexOf(" ");
+        int numStartIdx = spaceIdx + 1;
+        String removedInstructionString = trimmedInput.substring(numStartIdx);
+        String instructionString = trimmedInput.substring( 0, spaceIdx);
+        int taskNum;
+
+        if(instructionString.contains("unmark")){
+            taskNum = Integer.parseInt(removedInstructionString);
+            records[taskNum-1].setUndone();
+            printAllTasks();
+            return "";
+        }
+
+        if(instructionString.contains("mark")){
+            taskNum = Integer.parseInt(removedInstructionString);
+            records[taskNum-1].setDone();
+            printAllTasks();
+            return "";
+        }
+
+        createNewTask(instructionString, removedInstructionString);
+        return "";
+    }
+
+    private static void createNewTask(String instructionString, String removedInstructionString) {
+        Task task;
+        String taskDescription;
+        switch(instructionString) {
+        case "todo":
+            task = createToDo(removedInstructionString);
+            break;
+        case "deadline":
+            task = createDeadline(removedInstructionString);
+            break;
+        case "event":
+            task = createEvent(removedInstructionString);
+            break;
+        default:
+            return ;
+        }
+        addTaskToList(task);
+        printTaskAdded(task);
+        return ;
+    }
+
+    private static void printTaskAdded(Task task) {
+        printLine();
+        System.out.println("\tGot it. I've added this task:");
+        System.out.println("\t" + task);
+        System.out.println("\tNow you have " + recordsNum + " tasks in the list." );
+        printLine();
+    }
+
+    private static void addTaskToList(Task task) {
+        records[recordsNum] = task;
+        recordsNum++;
```

Comment:
you can merge these two line into one
```suggestion       
     records[recordsNum++] = task;
```